indexers/keyvalue/LevelDBIndexer/tests/test_leveldbindexer.py

```diff
@@ -96,118 +86,117 @@ def validate_results(save_abspath, results, negative_results):
             validate_negative_results(negative_results, searcher)
 
 
-def run_crud_test_for_scenario(indexer, actions, results, no_results):
-    exception = None
-    try:
-        with indexer:
-            indexer.touch()
-            indexer.save()
-        save_abspath = indexer.save_abspath
-        index_abspath = indexer.index_abspath
-        assert os.path.exists(index_abspath)
-        assert os.path.exists(save_abspath)
-        apply_actions(save_abspath, index_abspath, actions)
-        validate_results(save_abspath, results, no_results)
-    except Exception as e:
-        exception = e
-    finally:
-        rm_files([save_abspath, index_abspath])
-        if exception is not None:
-            raise exception
-
-
-def run_crud_test(actions, results, no_results):
+def run_crud_test_for_scenario(indexer, actions, results, no_results, tmpdir):
+    indexer.workspace = tmpdir
+    with indexer:
+        indexer.touch()
+        indexer.save()
+    save_abspath = indexer.save_abspath
+    index_abspath = indexer.index_abspath
+    assert os.path.exists(index_abspath)
+    assert os.path.exists(save_abspath)
+    apply_actions(save_abspath, index_abspath, actions)
+    validate_results(save_abspath, results, no_results)
+
+
+def run_crud_test(actions, results, no_results, tmpdir):
     # test construction from code
-    indexer = LevelDBIndexer(level='doc', index_filename='leveldb.db')
-    run_crud_test_for_scenario(indexer, actions, results, no_results)
+    indexer = LevelDBIndexer(level='doc', index_filename=os.path.join(tmpdir, 'leveldb.db'))
```

Comment:
use `pathlib` instead of os.path since we changed this in the core tests

Reply:
right, pathlib is more pleasant to use. Thanks